modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,333 @@
+            case PacketType::Event:
+                processEventPacket(packet);
+                break;
+
+            case PacketType::Data:
+                SAMPLE_TYPE_DISPATCH(inputSampleType, processDataPacket, packet);
+                break;
+
+            default:
+                break;
+        }
+
+        packet = connection.dequeue();
+    };
+}
+
+void ClassifierFbImpl::processEventPacket(const EventPacketPtr& packet)
+{
+    if (packet.getEventId() == event_packet_id::DATA_DESCRIPTOR_CHANGED)
+    {
+        DataDescriptorPtr inputDataDescriptor = packet.getParameters().get(event_packet_param::DATA_DESCRIPTOR);
+        DataDescriptorPtr inputDomainDataDescriptor = packet.getParameters().get(event_packet_param::DOMAIN_DATA_DESCRIPTOR);
+        processSignalDescriptorChanged(inputDataDescriptor, inputDomainDataDescriptor);
+    }
+}
+
+template <SampleType InputSampleType>
+void ClassifierFbImpl::processDataPacket(const DataPacketPtr& packet)
+{
+    using InputType = typename SampleTypeToType<InputSampleType>::Type;
+
+    if (packet.getSampleCount() == 0) {
+        return;
+    }
+    
+    //checking reaching of a new epoch
+    {
+        auto inputDomainData = static_cast<UInt*>(packet.getDomainPacket().getData());
+
+        UInt cur_time = inputDomainData[0];
+
+        if (packetStarted == UInt{}) {
+            packetStarted = cur_time;
+                sampleStarted = 0; 
+        }
+
+        auto diff = cur_time - packetStarted;
+            if (diff < 1000 * epochMs) {
+            packets.push_back(packet);
+            return;
+        }
+    }
+
+    if (packets.empty()) {
+        return;
+    }
+
+    auto rangeSize = outputDataDescriptor.getDimensions()[0].getSize();
+
+    auto outputDomainPacket = DataPacket(outputDomainDataDescriptor, 1);
+    auto outputDomainData = static_cast<UInt*>(outputDomainPacket.getData());
+
+    auto outputPacket = DataPacketWithDomain(outputDomainPacket, outputDataDescriptor, 1);
+    auto outputData = static_cast<InputType*>(outputPacket.getData());
+    memset(outputData, 0, rangeSize * sizeof(InputType));
+
+    // calculating end block for classifier
+    size_t lastPacketSamples = 1;
+    {   
+        auto& lastPacket = packets.back();
+        auto inputDomainData = static_cast<UInt*>(lastPacket.getDomainPacket().getData());
+        bool wasBreaked = false;
+
+        for(; lastPacketSamples < lastPacket.getSampleCount(); lastPacketSamples++) {
+            if (inputDomainData[lastPacketSamples] - packetStarted >= 1000 * epochMs) {
+                wasBreaked = true;
+                break;
+            }
+        }
+        lastPacketSamples += wasBreaked;
+
+        *outputDomainData = inputDomainData[lastPacketSamples - 1];
+        packetStarted += 1000 * epochMs;
+    }
+
+    Int offset = static_cast<Int>(outputDataDescriptor.getDimensions()[0].getRule().getParameters().get("start"));
+    double valCnt = 0;
+
+    for (const auto & list_packet: packets) {
```

Comment:
One input packet can contain huge number of samples, i.e. 1 million. That means that from one input packet multiple vector samples should be produced. I don't think this case is covered here.

Actually this would be a perfect use case for "block reader", but unfortunately we don't support readers in function blocks.